File: module_zhihu/src/main/java/com/gas/zhihu/main/mvp/MainContract.java
package com.gas.zhihu.main.mvp;

import android.app.Activity;
import android.content.Context;

import com.base.lib.mvp.IModel;
import com.base.lib.mvp.IView;
import com.gas.zhihu.bean.DailyListBean;

import java.util.List;

import io.reactivex.Observable;

public interface MainContract {


    interface View extends IView {
        Activity getActivity();
        void success();
        void onError();
    }


    interface Model extends IModel{
        Observable<DailyListBean> getDailyList();
    }

}

File: app/src/main/java/com/gas/app/ui/fragment/main/mvp/MainContract.kt
package com.gas.app.ui.fragment.main.mvp

import com.base.lib.mvp.IModel
import com.base.lib.mvp.IView

/**
 * ================================================
 * Description:
 *
 *
 * Created by GasMvpFragment on 11/30/2019 15:27
 * ================================================
 */
interface MainContract {
    //对于经常使用的关于UI的方法可以定义到IView中,如显示隐藏进度条,和显示文字消息
    interface View : IView

    //Model层定义接口,外部只需关心Model返回的数据,无需关心内部细节,即是否使用缓存
    interface Model : IModel
}